Samples/PDESampleApplication/Form2.cs

```diff
@@ -0,0 +1,319 @@
+        }
+
+        async void ProtectAndLog(IStorageItem item, UserDataAvailability level)
+        {
+            var protectResult = await g_udpm.ProtectStorageItemAsync(item, UserDataAvailability.AfterFirstUnlock);
+            if (protectResult == UserDataStorageItemProtectionStatus.Succeeded)
+            {
+                LogLine("Protected " + item.Name + " to level " + level);
+            }
+            else
+            {
+                LogLine("Protection failed for " + item.Name + " to level " + level + ", status: " + protectResult);
+            }
+        }
+
+        async void ProtectFolderRecursively(StorageFolder folder, UserDataAvailability level)
+        {
+            // Protect the folder first so new files / folders after this point will
+            // get protected automatically.
+            ProtectAndLog(folder, level);
+
+            // Protect all sub-folders recursively.
+            var subFolders = await folder.GetFoldersAsync();
+            foreach (var subFolder in subFolders)
+            {
+                ProtectFolderRecursively(subFolder, level);
+            }
+
+            // Finally protect all existing files in the folder.
+            var files = await folder.GetFilesAsync();
+            foreach (var file in files)
+            {
+                ProtectAndLog(file, level);
+            }
+        }
+
+        async void UnprotectBuffer(String g_protectbase64EncodedContent)
+        {
+            try
+            {
+                var protectedBuffer = CryptographicBuffer.DecodeFromBase64String(g_protectbase64EncodedContent);
+                var result = await g_udpm.UnprotectBufferAsync(protectedBuffer);
+                if (result.Status == UserDataBufferUnprotectStatus.Succeeded)
+                {
+                    LogLine("Result of Unprotecting the buffer:" +
+                        CryptographicBuffer.ConvertBinaryToString(BinaryStringEncoding.Utf8, result.UnprotectedBuffer));
+                    LogLine("Status of Unprotectng the buffer:" + result.Status);
+                }
+                else
+                {
+                    LogLine("This protected buffer is currently unavailable for unprotection");
+                }
+            }
+            catch(ArgumentException)
+            {
+                LogLine("Invalid Argument, please check text being unprotected!");
+            }
+            catch (Exception)
+            {
+                LogLine("Something went wrong");
+            }
+        }
+
+        async void ProtectBuffer(String text, UserDataAvailability level)
+        {
+            try
+            {
+                var buffer = CryptographicBuffer.ConvertStringToBinary(text, BinaryStringEncoding.Utf8);
+                g_protectedContent = await g_udpm.ProtectBufferAsync(buffer, level);
+                g_protectbase64EncodedContent = CryptographicBuffer.EncodeToBase64String(g_protectedContent);
```

Comment:
So we're passing the buffer that was protected last time and unprotecting that? Wouldn't it be better to let the user pass a custom protected buffer in the text box and unprotect that?

The point of this is that a buffer is protected by the app, persisted, and then unprotected at a later time - for at-rest encryption.

Letting the user provide the protected buffer allows simulating that. Keeping the last protected buffer is not intuitive.